server/src/main/java/io/crate/analyze/SubscriptValidator.java

```diff
@@ -40,10 +38,14 @@
 
 import java.util.Locale;
 
+/**
+ * NOTE: Array index out of bound validation cannot happen here since array index expressions such as
+ * `-1` or `x + 1` are not yet evaluated. The actual validation is handled by implicit cast of
+ * the array indexes to the target type (integer) of {@link io.crate.expression.scalar.SubscriptFunction}.
+ * See {@link io.crate.analyze.expressions.ExpressionAnalyzer#cast}
+ */
```

Comment:
@matriv, since negative expressions are another array indexes that cannot be validated here, it could be better to move the index range validation to the implicit cast altogether, WDYT?

Reply:
Hope my replies to other comments explains this as well. Please let me know. Thank you